x-pack/plugins/observability_solution/apm/public/components/app/service_tab_empty_state/index.tsx

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  EuiButton,
+  EuiButtonIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiImage,
+  EuiLink,
+  EuiPanel,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+  useEuiTheme,
+} from '@elastic/eui';
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { useApmPluginContext } from '../../../context/apm_plugin/use_apm_plugin_context';
+import defaultImgSrc from '../../../assets/service_tab_empty_state/service_tab_empty_state_default.png';
+
+export interface ServiceTabEmptyStateProps {
+  title: string;
+  content: string;
+  imgSrc: string | null;
+  dismissable?: boolean;
+  onDissmiss?: () => void;
+}
+
+const addAPMLinkButton = {
+  label: i18n.translate('xpack.apm.serviceTabEmptyState.addAPMButtonLabel', {
+    defaultMessage: 'Add APM',
+  }),
+  href: '/app/observabilityOnboarding/?category=apm',
```

Comment:
I know there are other places in the code where this URL is hardcoded but there's a [locator](https://github.com/cauemarcondes/kibana/blob/d8c85a3f06e2cb24bfaf8f15a587ddd6ae1dd8fd/x-pack/plugins/observability_solution/observability_onboarding/public/locators/onboarding_locator/locator_definition.ts#L21) we can use to avoid hardcoding it. You don't need to change it in the other places if you don't want to, but if you could update the new one, it would be nice. Thank you.

Reply:
There is a shared component which provides the url (hardcoded) https://github.com/elastic/kibana/blob/a0cc891a4acd1250ad284d89b1be5c056ae06d54/x-pack/plugins/observability_solution/apm/public/components/shared/add_data_buttons/buttons.tsx#L34 

if we can update the hardcoded url with the locator and use `addApmData.link` it would be great

Reply:
I'll make the change. Thank you both!

Reply:
@kpatticha, I'm unsure about using `addApmData.link`. 

While it's part of a shared component, it feels odd to use constants created specifically for that shared component in other components. Should we be doing this?

Reply:
We could use the shared button component. 

https://github.com/elastic/kibana/blob/a0cc891a4acd1250ad284d89b1be5c056ae06d54/x-pack/plugins/observability_solution/apm/public/components/shared/add_data_buttons/buttons.tsx#L37C17-L56

Reply:
Alright, I'll use `AddApmData` as suggested and try to update the `href` to use the locator.